### PR TITLE
Added support for platform storage. Refactored error handling.

### PR DESCRIPTION
**Pull request by jbasdf**

### CODE REVIEW

File: lib/atomic_lti/open_id_middleware.rb [hunk 1]
```diff
@@ -146,46 +151,33 @@ def matches_target_link?(request)
...
       env["atomic.validated.id_token"] = id_token
 
+      #platform = AtomicLti::Platform.find_by!(iss: id_token["iss"])
+      platform = AtomicLti::Platform.find_by!(iss: id_token_decoded["iss"])
+      if request.params["lti_storage_target"].present? && AtomicLti.use_post_message_storage
+        lti_storage_params = build_lti_storage_params(request, platform)
+        # Add the values needed to do client side validate to the environment
+        env["atomic.validated.state_verified"] = state_verified
+        env["atomic.validated.state"] = state
+        env["atomic.validated.lti_storage_params"] = lti_storage_params
```
Comment (mpetrowi) [inline lib/atomic_lti/open_id_middleware.rb:177]: We could wrap them all together:
```
 env["atomic.validated.state_validation"] = { state: state, lti_storage_params: lti_storage_params, verified_by_cookie: state_verified }
```
And then the render code in the app just has to make that one thing available.

Comment (jbasdf) [reply lib/atomic_lti/open_id_middleware.rb:177]: I changed it

File: lib/atomic_lti/open_id_middleware.rb [hunk 2]
```diff
@@ -69,59 +73,61 @@ def handle_init(request)
...
           launch_url: target_link_uri,
-          settings: {
-            require_csrf: AtomicLti.enforce_csrf_protection,
-            state: request.params["state"],
-            lti_storage_params: lti_storage_params,
-          },
         },
       )
 
       [200, { "Content-Type" => "text/html" }, [html]]
```
Comment (mpetrowi) [inline lib/atomic_lti/open_id_middleware.rb:124]: We could delete the state cookie here.

**Review by mpetrowi — APPROVED** [review]
Looks great!

**Pull request merged**